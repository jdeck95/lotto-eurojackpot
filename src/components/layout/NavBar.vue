<script setup lang="ts"></script>

<template>
  <q-toolbar class="toolbar text-white justify-center">
    <div class="row no-wrap">
      <img
        src="@/assets/wow-logo.png"
        alt="Wheel Of Wealth Logo"
        width="140px"
        hight="100px"
      />
      <div class="column justify-center title q-mr-lg">Wheel Of Wealth</div>
    </div>
    <router-link
      class="q-mr-md"
      to="/"
      >Start</router-link
    >
    <router-link to="/eurojackpot">Eurojackpot</router-link>
  </q-toolbar>
</template>

<style scoped lang="scss">
.toolbar {
  background-color: #333a65;
}

a {
  text-decoration: none;
  font-size: 20px;
  color: white;
  padding: 12px;
  border-radius: 3px;

  &:hover {
    text-decoration: underline;
  }
}

.router-link-active {
  background-color: #b7abb6;
  color: black;
  padding: 12px;
  text-decoration: underline;
}

.title {
  font-size: 26px;
  margin-left: -20px;
}

@media screen and (max-width: $breakpoint-xs) {
  .title {
    display: none;
  }
}
</style>
