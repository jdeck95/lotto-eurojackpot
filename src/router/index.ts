import { createRouter, createWebHistory } from 'vue-router';
import HomeView from '@/views/HomeView.vue';
import MainLayout from '@/layout/MainLayout.vue';

const router = createRouter({
  history: createWebHistory(import.meta.env.BASE_URL),
  routes: [
    {
      path: '/',
      name: 'home',
      meta: { layout: MainLayout },
      component: HomeView,
    },
    {
      path: '/eurojackpot',
      name: 'eurojackpot',
      meta: { layout: MainLayout },
      component: () => import('../views/EurojackpotView.vue'),
    },
  ],
});

export default router;
