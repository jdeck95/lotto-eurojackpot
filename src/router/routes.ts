import HomeView from '@/views/HomeView.vue';
import MainLayout from '@/layout/MainLayout.vue';

export const routes = [
  {
    path: '/',
    name: 'home',
    label: 'Start',
    meta: { layout: MainLayout },
    component: HomeView,
  },
  {
    path: '/eurojackpot',
    name: 'eurojackpot',
    label: 'Eurojackpot',
    meta: { layout: MainLayout },
    component: () => import('../views/EurojackpotView.vue'),
  },
];
