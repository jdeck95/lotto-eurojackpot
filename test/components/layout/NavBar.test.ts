import { mount, type VueWrapper } from '@vue/test-utils';
import { createTestingPinia } from '@pinia/testing';
import { setActivePinia } from 'pinia';
import { Quasar } from 'quasar';
import { beforeEach, describe, expect, it } from 'vitest';
import NavBar from '../../../src/components/layout/NavBar.vue';
import router from '@/router';

describe('NavBar', () => {
  let wrapper: VueWrapper<any>;

  const pinia = createTestingPinia();
  setActivePinia(pinia);

  beforeEach(() => {
    wrapper = mount(NavBar, {
      global: {
        plugins: [Quasar, pinia, router],
      },
    });
  });

  it('should render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
