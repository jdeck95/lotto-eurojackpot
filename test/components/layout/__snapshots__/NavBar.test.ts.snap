// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavBar > should render correctly 1`] = `
"<div data-v-d34746f4="" class="q-toolbar row no-wrap items-center toolbar text-white justify-center" role="toolbar">
  <div data-v-d34746f4="" class="row no-wrap"><img data-v-d34746f4="" src="/src/assets/wow-logo.png" alt="Wheel Of Wealth Logo" width="140px" hight="100px">
    <div data-v-d34746f4="" class="column justify-center title q-mr-lg">Wheel Of Wealth</div>
  </div><a data-v-d34746f4="" href="/" class="q-mr-md">Start</a><a data-v-d34746f4="" href="/eurojackpot" class="q-mr-md">Eurojackpot</a>
</div>"
`;
